bnm/recon/algo/service/volume.py

```diff
@@ -226,97 +232,107 @@ def mask_to_vol(self, in_vol_path, mask_vol_path, out_vol_path=None, labels=None
+        IOUtils.write_volume(out_volume_path, tdi_volume)
 
-        self.volume_io.write(out_fname, tdi_volume)
-
-    # It removes network nodes with zero connectivity, and returns a symmetric connectivity matrix
-    # Inputs:
-    #    - the tdi_lbl.nii volume path
-    #    -a .csv connectivity file path, output of Mrtrix3 tck2connectome
-    # Outputs:
-    #   - the symmetric no-zero-connections connectivity matrix saved as .npy
-    #   - the tdi_lbl.nii volume with the removed voxel nodes to 0 and the labels
-    #    updated
-    # Optionally: if the tract length matrix is in the input, it is also processed
     def remove_zero_connectivity_nodes(self, node_volume_path, connectivity_matrix_path, tract_length_path=None):
-        node_volume = self.volume_io.read(node_volume_path)
-        out_volume = Volume(node_volume.data, node_volume.affine_matrix, node_volume.header)
+        """
+        It removes network nodes with zero connectivity from the volume and connectivity matrices.
+        The zero connectivity nodes will be labeled with 0 in the volume and the remaining labels will be updated.
+        The connectivity matrices will be symmetric.
+        :param node_volume_path: tdi_lbl.nii volume path
+        :param connectivity_matrix_path: .csv file, output of Mrtrix3 tck2connectome
+        :param tract_length_path: optional .csv tract lengths matrix
+        :return: overwrites the input volume and matrices with the processed ones. Also saves matrices as .npy.
+        """
+
+        node_volume = IOUtils.read_volume(node_volume_path)
 
         connectivity = numpy.array(numpy.genfromtxt(connectivity_matrix_path, dtype='int64'))
-
-        # Make it symmetric:
         connectivity = connectivity + connectivity.T
-
         connectivity_row_sum = numpy.sum(connectivity, axis=0)
-        nodes_to_keep_indices = connectivity_row_sum > 0
 
-        # Select only the specified columns and rows from the connectivity matrix:
+        nodes_to_keep_indices = connectivity_row_sum > 0
         connectivity = connectivity[nodes_to_keep_indices, :][:, nodes_to_keep_indices]
 
-        numpy.save(os.path.splitext(connectivity_matrix_path)[0] + ".npy", connectivity)
-        numpy.savetxt(connectivity_matrix_path, connectivity)
+        numpy.save(os.path.splitext(connectivity_matrix_path)[0] + NPY_EXTENSION, connectivity)
+        numpy.savetxt(connectivity_matrix_path, connectivity, fmt='%1d')
 
         if os.path.exists(str(tract_length_path)):
             connectivity = numpy.array(numpy.genfromtxt(tract_length_path, dtype='int64'))
             connectivity = connectivity[nodes_to_keep_indices, :][:, nodes_to_keep_indices]
 
-            numpy.save(os.path.splitext(tract_length_path)[0] + ".npy", connectivity)
-            numpy.savetxt(tract_length_path, connectivity)
+            numpy.save(os.path.splitext(tract_length_path)[0] + NPY_EXTENSION, connectivity)
+            numpy.savetxt(tract_length_path, connectivity, fmt='%1d')
```

Comment:
Instead of these many numpy.save[txt] sparsely spread along this method, wouldn't be better to have a 'write' method in the IO module?